servers/rendering/renderer_scene_render.h

```diff
@@ -44,27 +45,81 @@ class RendererSceneRender {
 	};
 
 	struct GeometryInstance {
+		// setup
```

Comment:
I don't think I like implementation detail being exposed. I think what Cull sees should be entirely pure virtual, specially because for GDExtension we will need to re-route everything to GDVIRTUAL eventually.

IMO I would leave all this pure virtual and if you want to have all this shared, do it in a class that inherits GeometryInstance before being inherited for forward clustered or mobile.

Reply:
We need some sort of encapsulation level to eventually expose these things to the binder.

Reply:
@reduz I'm happy to put a level inbetween though seeing most of this state will be duplicated for any implementation, doesn't it make more sense to add getters alongside the setters so both getters and setters can be exposed to a binder library?

For making it possible to implement such object through the binder, I guess at some point we should just turn GeometryInstance into an proper object?

Anyway I'll change it to the way you suggest :)